testapi.pm

```diff
@@ -2313,4 +2315,24 @@ sub compat_args {
     return %ret;
 }
 
+=head2 show_curl_progress_meter
+
+Helper function to alter the curl command to show progress meter.
+This works only when uploading where the output is not lately used.
+
+    show_curl_progress_meter( $cmd )
+
+A typical call would look like:
+
+    $cmd .= show_curl_progress_meter($cmd);
+=cut
+sub show_curl_progress_meter {
+    my $cmd = shift;
+
+    # Redirect the output so the progress meter is displayed 
+    $cmd .= "-o /dev/$serialdev " if get_var('UPLOAD_METER');
+
+    return $cmd;
+}
```

Comment:
with this you are adding `$cmd` twice. Ok, I think it's clearer if we let the function just return necessary additional parameters. How about just

```suggestion
# Redirect the output so the progress meter is displayed
sub show_curl_progress_meter { "-o /dev/$serialdev " if get_var('UPLOAD_METER') }
```

and then use like this:

```perl
$cmd .= show_curl_progress_meter;
```